tracing-core/src/span.rs

```diff
@@ -68,12 +68,28 @@ impl Id {
         Id(NonZeroU64::new(u).expect("span IDs must be > 0"))
     }
 
+    /// Constructs a new span ID from the given `NonZeroU64`.
+    ///
+    /// Unlike [`Id::from_u64`](#method.from_u64), this will never panic.
+    #[inline]
```

Comment:
Not convinced we _need_ inline attributes for these --- I can't imagine them ever not being inlined. Not a blocker.

Reply:
I noticed the lack of `#[inline]` in this crate. In my experience, not having the attribute results in a function call, which is needless overhead. However, I haven't tested whether this is the case for newer compilers. I remember seeing a compiler issue tracking this behavior.